core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -515,6 +515,7 @@ class TransactionStateManager(brokerId: Int,
         getTransactionState(transactionalId) match {
 
           case Left(err) =>
+            info(s"TransactionalId $transactionalId : failing log append of $newMetadata due to $err.")
```

Comment:
As @hachikuji pointed out, we are should not call `responseCallback(err)` in the next line but just set `responseError = err`. 

I think using info is okay, as we use info on other error branches below as well.